igor-web/src/main/java/com/netflix/spinnaker/igor/scm/AbstractScmMaster.java

```diff
@@ -32,4 +32,8 @@ public String getTextFileContents(
   public String getTextFileContents(String projectKey, String repositorySlug, String path) {
     return getTextFileContents(projectKey, repositorySlug, path, ScmMaster.DEFAULT_GIT_REF);
   }
+
+  public Object getCommitDetails(String projectKey, String repositorySlug, String commitSha) {
```

Comment:
Could this return a `Commit` object instead?

Reply:
problem is, `Commit` is implemented per SCM, so there is no generic struct to return here unless we create an abstract one. Alternatively we can try to consolidate them all into something more generic, but that appeared to be a chore we can deal with independent of this PR.

Reply:
What about using a `GenericGitRevision`? That more or less corresponds to a commit.

Reply:
@nimakaviani I see. I'm OK tackling this in a separate PR.

Reply:
@jervi `GenericGitRevision` is close, but `commit` and `author` info returned from different SCMs (github or gitlab) are more complex data structures that do not match one-to-one to the stringified values in `GenericGitRevision`, unfortunately.